src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -149,140 +154,80 @@ def _lint_queries(self, context: _ReportingContext) -> None:
             tables = self.collect_used_tables_from_query("no-dashboard-id", query)
             context.all_tables.extend(tables)
 
-    def _dashboard_ids_in_scope(self) -> list[str]:
-        if self._include_dashboard_ids is not None:  # an empty list is accepted
-            return self._include_dashboard_ids
-        items_listed = 0
-        dashboard_ids = []
-        # redash APIs are very slow to paginate, especially for large number of dashboards, so we limit the listing
-        # to a small number of items in debug mode for the assessment workflow just to complete.
-        for dashboard in self._ws.dashboards.list():
-            if self._debug_listing_upper_limit is not None and items_listed >= self._debug_listing_upper_limit:
-                logger.warning(f"Debug listing limit reached: {self._debug_listing_upper_limit}")
-                break
-            if dashboard.id is None:
-                continue
-            dashboard_ids.append(dashboard.id)
-            items_listed += 1
-        return dashboard_ids
-
-    def _queries_in_scope(self) -> list[LegacyQuery]:
-        if self._include_dashboard_ids is not None:  # an empty list is accepted
-            return []
-        items_listed = 0
-        legacy_queries = []
-        for query in self._ws.queries_legacy.list():
-            if self._debug_listing_upper_limit is not None and items_listed >= self._debug_listing_upper_limit:
-                logger.warning(f"Debug listing limit reached: {self._debug_listing_upper_limit}")
-                break
-            legacy_queries.append(query)
-            items_listed += 1
-        return legacy_queries
+    def _list_queries(self) -> Iterable[Query]:
+        for crawler in self._dashboard_crawlers:
+            yield from crawler.list_queries()
```

Comment:
Ah, now I understand.

Reply:
Yeah, similarly to the notebook and such, we do not persist those resources in the inventory. (I think we should not do that.) However, that means we need to retrieve the code resources dynamically, thus I added this public method